src/v2/services/implementations/AssetsService.ts

```diff
@@ -38,23 +39,27 @@ export class AssetsService implements IAssetsService {
   }
 
   public async transferNativeAsset(param: ParamAssetTransfer): Promise<void> {
-    const useableBalance = await this.AssetsRepository.getNativeBalance(param.senderAddress);
-    const isBalanceEnough =
-      Number(ethers.utils.formatEther(useableBalance)) -
-        Number(ethers.utils.formatEther(param.amount)) >
-      REQUIRED_MINIMUM_BALANCE;
+    const isNativeToken = param.assetId === idAstarNativeToken;
+    // Memo: Check if the native token's remaining balance is enough to pay the transaction fee
+    if (isNativeToken) {
+      const useableBalance = await this.AssetsRepository.getNativeBalance(param.senderAddress);
+      const isBalanceEnough =
+        Number(ethers.utils.formatEther(useableBalance)) -
```

Comment:
There are unnecessary conversions here. The main reason for that is `useableBalance` and `param.amount` are strings. IMO using string for balances is not natural (more natural is `bigint`) and leads to increased code complexity.
You don't need to fix this now, but we should have this in mind for the future

Reply:
Noted, I've modified like [this](https://github.com/AstarNetwork/astar-apps/pull/1429/commits/a60c78c9ebd58c736b53cd18e81b81967df0d974) but we need to do some refactoring in the future.